Add node.js v12 on Alpine v3.13/v3.14

DaKaZ opened the pull request:
## Description

This PR add support for Node v12 on Alpine 3.13

## Motivation and Context

Node v12 is still in LTS support and Alpine 3.13 has numerous security improvements, including accessing packages with `apk` via HTTPS instead of HTTP

## Testing Details

I have been running v12 on this docker image for two production containers for the last 2 weeks

## Example Output(if appropriate)

## Types of changes

* Alpine 3.13 + Alpline 3.14 for Node 12
* Bumped NodeJS version to 12.22.1

- [ ] Documentation
- [x] Version change (Update, remove or add more Node.js versions)
- [x] Variant change (Update, remove or add more variants, or versions of variants)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Others (non of above)

## Checklist

<!--
Go over all the following points, and put an `x` in all the boxes that apply.
If you're unsure about any of these, don't hesitate to ask. We're here to help!
-->

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING.md** document.
- [x] All new and existing tests passed.

ChuanyuWang commented:
This pull request is highly wanted. We also have customers running on top of Nodejs 12, but facing vulnerabilities from Alpine 3.12

SimenB commented:
@nodejs/docker any reason not to do this? v12 won't be EOL for another year

LaurentGoderre commented:
I think it's relatively low risk. @tianon what do you think?

We usually don't add old alpine versions to new node version but we didn't have a request yet to add a new alpine version to an older version of Node.

LaurentGoderre commented:
@DaKaZ could you fix the conflict?

tianon commented:
Given that things are appropriately tagged with specific Alpine versions as aliases, any users broken by adjusting the default have a pretty easy escape hatch (esp. since there's no removal of any versions yet).

In repos like PHP where each new underlying version adds a lot of overhead (build time, variant sprawl), we tend to limit to the latest two for Alpine, so something like that might be worth adopting here.

Marcuse7 commented:
Merci @DaKaZ and @LaurentGoderre for your work! I would also appreciate very much if this PR was reviewed/merged. We have legacy software running on LTS version that we cannot migrate for now, and would love to see it fixed.

Marcuse7 reviewed (approved):
Looks okay. I see no conflicting changes in _12/architectures_ either.

LaurentGoderre commented:
I rebased manually. @nschonni I am a bit out of touch, is this enough?

nschonni commented:
@LaurentGoderre It would probably need something added in https://github.com/nodejs/docker-node/blob/cdd626c9e138b670318c56a39a5c86b0f5bb271b/versions.json#L186

LaurentGoderre commented on versions.json:
```diff
@@ -222,6 +222,14 @@
+        "arm32v7",
+        "arm64v8",
+        "ppc64le",
+        "s390x"
```
@nschonni like this?

Marcuse7 replied:
I am not nschonni but here's my 2 cts. Pls use spaces instead of tabs. The _eclint_ checker had reported: 
`202:01 ❌  invalid indent style: tab, expected: space`

DaKaZ replied:
This should be fixed

nschonni commented on versions.json:
```diff
@@ -222,6 +222,14 @@
         "ppc64le",
         "s390x"
+      ],
+	  "alpine3.13": [
```
Yeah, I think that should be it, just spacing
```suggestion
     "alpine3.13": [
```

DaKaZ replied:
This is fixed :)

wayne-li2 commented:
Thanks for the backport! Some of our containers have Python `cryptography` dependencies, and the Rust version on Alpine 3.12 is too low. We are building Rust from source right now but it's a lot cleaner if we can just `apk add rust cargo`, which hopefully we can in 3.13.

PeterDaveHello commented on 12/alpine3.13/Dockerfile:
```diff
@@ -0,0 +1,100 @@
+  && node --version \
+  && npm --version
+
+ENV YARN_VERSION 1.22.10
```
This one might need to align with the others, to be 1.22.5?

PeterDaveHello commented on 12/alpine3.13/Dockerfile:
```diff
@@ -0,0 +1,100 @@
+FROM alpine:3.13
+
+ENV NODE_VERSION 12.20.2
```
Also this version number here need to update again.

PeterDaveHello reviewed (changes requested):
Please use the update script to update the Dockerfile to be the latest status.

kibertoad commented:
Maybe this should skip 3.13 and go straight to 3.14 now that it was released?

PeterDaveHello commented:
Or maybe just add them both.

DaKaZ commented:
@PeterDaveHello and others - sorry for the delay, got busy with some other things.  I have update the node version, fixed the indentation issue, and added support for Alpine 3.14.  Lets land this :)

PeterDaveHello commented:
Oops, a little bit embarrassing here, as #1520 #1521 security updates got merged, @DaKaZ do you mind to do it again? I'll get it reviewed and merged ASAP.

DaKaZ commented:
@PeterDaveHello no problem at all, I just updated the PR

PeterDaveHello commented:
Thanks @DaKaZ
